src/Xrefcheck/Verify.hs

```diff
@@ -359,6 +364,62 @@ checkExternalResource VerifyConfig{..} link
           Nothing -> False
         Left _ -> False
 
+    analyzeURI :: ExceptT VerifyError IO URI
+    analyzeURI =
+      let initialMkURIRequest = mkURI link :: Maybe URI
+          bracketIndices = searchBrackets link
+      in case initialMkURIRequest of
+        Just uri -> pure uri
+        Nothing -> throwError . ExternalResourceInvalidUri $
+          if null bracketIndices
+          then Nothing
+          else Just $ invalidURIVerbose bracketIndices
+
+    invalidURIVerbose :: [Int] -> Text
+    invalidURIVerbose bracketIndices = fmt . indentF 3 . unlinesF $
```

Comment:
I’m confused. I was going to comment here that instead of referring to RFC, we should refer to WHATWG URL, and also instead of `bracketIndices` and `unexpected bracket` it should be more general for any invalid characters. But then, @dcastro’s screenshot in Slack shows this fixed. Is there some other version of this change?

Reply:
Anyway, what I wanted to say is: let’s just get rid of the term URI and always use URL in all user-visible parts.

Reply:
> @dcastro’s screenshot in Slack shows this fixed. Is there some other version of this change?

That was my mistake, I had forgotten to push the latest changes to this PR I had made long before @dcastro's review of the old version. The old version (@dcastro's screenshot) did indeed contain the WHATWG URL reference, which I replaced with the RFC reference in the recent `push`. 

> let’s just get rid of the term URI and always use URL in all user-visible parts

> instead of bracketIndices and unexpected bracket it should be more general for any invalid characters

Got it.

Reply:
Ok, I see. I thought that the WHATWG reference was actually better, since the RFC is pretty ambiguous on that matter and WHATWG URL, in essence, deprecates it anyway. So I suggest returning WHATWG URL.